pkg/apis/verify/v1beta1/certificaterequest_types.go

```diff
@@ -30,14 +30,15 @@ type CertificateAuthoritySpec struct {
 // CertificateRequestSpec defines the desired state of CertificateRequest
 // +k8s:openapi-gen=true
 type CertificateRequestSpec struct {
-	CountryCode          string                    `json:"countryCode,omitempty"`
-	CommonName           string                    `json:"commonName"`
-	ExpiryMonths         int                       `json:"expiryMonths,omitempty"`
-	Location             string                    `json:"location,omitempty"`
-	Organization         string                    `json:"organization,omitempty"`
-	OrganizationUnit     string                    `json:"organizationUnit,omitempty"`
-	CACert               bool                      `json:"CACert,omitempty"`
-	CertificateAuthority *CertificateAuthoritySpec `json:"certificateAuthority,omitempty"`
+	CountryCode            string                    `json:"countryCode,omitempty"`
+	CommonName             string                    `json:"commonName"`
+	ExpiryMonths           int                       `json:"expiryMonths,omitempty"`
+	Location               string                    `json:"location,omitempty"`
+	Organization           string                    `json:"organization,omitempty"`
+	OrganizationUnit       string                    `json:"organizationUnit,omitempty"`
+	CACert                 bool                      `json:"CACert,omitempty"`
+	CertificateAuthority   *CertificateAuthoritySpec `json:"certificateAuthority,omitempty"`
+	RegenerateWhenDaysLeft *int                      `json:"regenerateWhenDaysLeft,omitempty" hash:"ignore"`
```

Comment:
Why is this one a pointer to an int, whereas others (e.g. ExpiryMonths) are just straight ints?

Reply:
It's because of the way Golang handles nil values. Basically we want to be able to differentiate between if this field is unset (meaning we don't want to automatically regenerate ever) and if it's set to 0 (not very likely but not impossible).
If this fields type was just `int` and we didn't define the field Go would set it to the types zero value, which for int's is...`0`. As it's a pointer, Go will set it to a pointers zero value, `nil`. This we can check of the value is `nil` to determine if it was not defined.

Reply:
Ah, OK. I see someone made an `Optional` thing, but it's probably overkill to bring that in just for one case. https://github.com/markphelps/optional